tests/scenario/test_gateway.py

```diff
@@ -343,3 +346,69 @@ def _get_listener_given_name(gateway, name: str):
         if listener["name"] == name:
             return listener
     raise KeyError(f"Listener with name {name} not found")
+
+
+def test_construct_hpa(istio_ingress_charm, istio_ingress_context):
+    """Assert that the HPA definition is constructed as expected."""
+    with istio_ingress_context(
+        istio_ingress_context.on.update_status(),
+        state=scenario.State(leader=True, planned_units=3),
+    ) as manager:
+        charm = manager.charm
+
+        hpa = charm._construct_hpa(charm.model.app.planned_units())
+        assert hpa.metadata.name == charm.app.name
+        assert hpa.metadata.namespace == charm.model.name
+
+        spec = hpa.spec
+        assert spec.minReplicas == charm.model.app.planned_units()
+        assert spec.maxReplicas == charm.model.app.planned_units()
+
+        ref = spec.scaleTargetRef
+        assert ref.apiVersion == "apps/v1"
+        assert ref.kind == "Deployment"
+        assert ref.name == "istio-ingress-k8s-istio"
```

Comment:
nit: `charm.model.app.planned_units()` feels a little bit like using an implementation detail as part of the test expectations.  99.9% sure it works as you expect, but imo we should do this:

```suggestion
    n_units = 3
    model_name = "test-model"
    with istio_ingress_context(
        istio_ingress_context.on.update_status(),
        state=scenario.State(leader=True, planned_units=n_units, model=Model(name=model_name)),
    ) as manager:
        charm = manager.charm

        hpa = charm._construct_hpa(n_units)
        assert hpa.metadata.name == istio_ingress_context._app_name
        assert hpa.metadata.namespace == model_name

        spec = hpa.spec
        assert spec.minReplicas == n_units
        assert spec.maxReplicas == n_units

        ref = spec.scaleTargetRef
        assert ref.apiVersion == "apps/v1"
        assert ref.kind == "Deployment"
        assert ref.name == "istio-ingress-k8s-istio"
```